pkg/scheduler/core/generic_scheduler.go

```diff
@@ -190,3 +191,24 @@ func (g *genericScheduler) chooseSpreadGroup(spreadGroup *util.SpreadGroup) []st
 	}
 	return feasibleClusters
 }
+
+func (g *genericScheduler) assignReplicas(clusters []*clusterv1alpha1.Cluster, replicaSchedulingStrategy *policyv1alpha1.ReplicaSchedulingStrategy, object *workv1alpha1.ObjectReference) ([]workv1alpha1.TargetCluster, error) {
+	if len(clusters) == 0 {
+		return nil, fmt.Errorf("no clusters available to schedule")
+	}
+	targetClusters := make([]workv1alpha1.TargetCluster, len(clusters))
+
+	if object.Replicas > 0 && replicaSchedulingStrategy != nil {
+		if replicaSchedulingStrategy.ReplicaSchedulingType == policyv1alpha1.ReplicaSchedulingTypeDuplicated {
+			for i, cluster := range clusters {
+				targetClusters[i] = workv1alpha1.TargetCluster{Name: cluster.Name, Replicas: object.Replicas}
+			}
+			return targetClusters, nil
+		}
+	}
+
+	for i, cluster := range clusters {
+		targetClusters[i] = workv1alpha1.TargetCluster{Name: cluster.Name}
+	}
```

Comment:
Does this always make a copy of `clusters`? for non-replica resources?

Reply:
Yes